src/tests/nativeaot/SmokeTests/DwarfDump/Program.cs

```diff
@@ -0,0 +1,71 @@
+using System;
+using System.Diagnostics;
+using System.IO;
+using System.Runtime.InteropServices;
+
+public class Program
+{
+    public static int Main(string[] args)
+    {
+        var llvmDwarfDumpPath = Path.Combine(
+            Environment.GetEnvironmentVariable("CORE_ROOT"),
+            "SuperFileCheck",
+            "runtimes",
+            RuntimeInformation.RuntimeIdentifier,
+            "native",
+            "llvm-dwarfdump");
+
+        if (!File.Exists(llvmDwarfDumpPath))
+        {
+            // Linux-only test
+            Console.WriteLine("Could not find llvm-dwarfdump at path: " + llvmDwarfDumpPath);
+            return 1;
+        }
+
+        Console.WriteLine("Running llvm-dwarfdump");
+        var proc = Process.Start(llvmDwarfDumpPath, "--version");
+
+        if (proc is null)
+        {
+            Console.WriteLine("llvm-dwarfdump could not run");
+            return 2;
+        }
+
+        proc.WaitForExit();
+        if (proc.ExitCode != 0)
+        {
+            return 3;
+        }
+
+        proc = Process.Start(new ProcessStartInfo
+        {
+            FileName = "/bin/sh",
+            Arguments = $"-c \"{llvmDwarfDumpPath} --verify {Environment.ProcessPath}",
+            RedirectStandardOutput = true,
+            RedirectStandardError = true,
+            UseShellExecute = false
+        });
+
+        // Just count the number of warnings and errors. There are so many right now that it's not worth enumerating the list
+        const int MinWarnings = 18400;
+        const int MaxWarnings = 48600;
+        int count = 0;
+        string line;
+        while ((line = proc.StandardOutput.ReadLine()) != null)
+        {
+            if (line.Contains("warning:") || line.Contains("error:"))
+            {
+                count++;
+            }
+        }
+        proc.WaitForExit();
+        if (count is not >= MinWarnings and <= MaxWarnings)
+        {
+            Console.WriteLine($"Found {count} warnings and errors, expected between {MinWarnings} and {MaxWarnings}");
+            Console.WriteLine("This is likely a result of debug info changes. To see the new output, run the following command:");
+            Console.WriteLine("\tllvm-dwarfdump --verify " + Environment.ProcessPath);
+            return 1;
```

Comment:
Would it be possible to have this print the stdout and stderr of the `--verify` we just ran so that we can see the detail of the problem?

I'm just imagining myself in the shoes of the person seeing this failure and saying "great, do I need to spin up a Linux machine to get a repro of this?".

Reply:
Might also be useful to print the `count` all the time so that if it fails, one can go to a different PR and see if the count already was very close to the threshold and the fix is to just update the range.

Basically, if I can fix a test failure by looking at logs, I'm happier than if I have to repro the failure locally first.

Reply:
I think output the log makes a ton of sense -- but only when the log is smaller than 50000 lines. I'm happy to make that change when we get the warnings down to a reasonable number.

Always printing the count is a great idea, though.